crates/e2e/tests/e2e/protocol_fee.rs

```diff
@@ -38,6 +41,263 @@ async fn local_node_price_improvement_fee_sell_order() {
+    );
+
+    let market_price_improvement_uid = services
+        .create_order(&market_price_improvement_order)
+        .await
+        .unwrap();
+    let limit_surplus_order_uid = services.create_order(&limit_surplus_order).await.unwrap();
+    let partner_fee_order_uid = services.create_order(&partner_fee_order).await.unwrap();
+
+    let mut config = vec![
+        "--drivers=test_solver|http://localhost:11088/test_solver".to_string(),
+        "--price-estimation-drivers=test_quoter|http://localhost:11088/test_solver".to_string(),
+    ];
+    config.extend(autopilot_config);
+    services.start_autopilot(None, config);
+
+    tracing::info!("Acquiring a quote.");
+    let quote_request = OrderQuoteRequest {
+        sell_token: onchain.contracts().weth.address(),
+        buy_token: limit_order_token.address(),
+        side: OrderQuoteSide::Sell {
+            sell_amount: SellAmount::AfterFee {
+                value: NonZeroU256::try_from(5).unwrap(),
+            },
+        },
+        validity: Validity::To(model::time::now_in_epoch_seconds() + 300),
+        ..Default::default()
+    };
+    let quote = services.submit_quote(&quote_request).await.unwrap();
+
+    tracing::info!("Rebalancing AMM pool.");
+    onchain
+        .mint_token_to_weth_uni_v2_pool(&limit_order_token, to_wei(1000))
+        .await;
+
+    tracing::info!("Waiting for liquidity state to update");
+    wait_for_condition(TIMEOUT, || async {
+        // Mint blocks until we evict the cached liquidity and fetch the new state.
+        onchain.mint_block().await;
+        let next = services.submit_quote(&quote_request).await.unwrap();
+        next.quote.buy_amount != quote.quote.buy_amount
+    })
+    .await
+    .unwrap();
+
+    tracing::info!("Waiting for trade.");
+    wait_for_condition(TIMEOUT, || async { services.solvable_orders().await == 3 })
+        .await
+        .unwrap();
+
+    wait_for_condition(TIMEOUT, || async { services.solvable_orders().await == 0 })
+        .await
+        .unwrap();
+
+    tracing::info!("Waiting for orders metadata update.");
+    let metadata_updated = || async {
+        onchain.mint_block().await;
+        let order_a = services
+            .get_order(&market_price_improvement_uid)
+            .await
+            .unwrap();
+        let order_b = services.get_order(&limit_surplus_order_uid).await.unwrap();
+        let order_c = services.get_order(&partner_fee_order_uid).await.unwrap();
+        !order_a.metadata.executed_surplus_fee.is_zero()
+            && !order_b.metadata.executed_surplus_fee.is_zero()
+            && !order_c.metadata.executed_surplus_fee.is_zero()
+    };
+    wait_for_condition(TIMEOUT, metadata_updated).await.unwrap();
+
+    let market_price_improvement = services
+        .get_order(&market_price_improvement_uid)
+        .await
+        .unwrap();
+    assert_approximately(
+        market_price_improvement.metadata.executed_surplus_fee,
+        202972993230628u128.into(),
```

Comment:
Are we actually seeing any price improvement here? Nothing about the pool changed, so this order should be executed as expected. I think we should:

1. Get a quote for the trade and use that as limit price (expected price)
2. Place the order
3. Inject more liquidity into the pool
4. Get another quote (now with expected price improvement).

Assert that `executed_surplus_fee` > 1/2 of the difference in the two quotes